File: src/lib/types/image.ts
export interface AiDetectionResult {
  is_ai_generated: boolean;
  reasons: string[];
  exif_data: Record<string, string> | null;
  format: string;
  png_metadata: Record<string, string> | null;
  has_c2pa: boolean;
  c2pa_manifest: string | null;
}

export interface ImageMetadata {
  image_data: string;
  exif_data: string;
  width: number;
  height: number;
  aspect_ratio: string;
  format: string;
  color_depth: number | null;
  file_size: number;
}

File: src-tauri/src/models/image.rs
use serde::Serialize;

#[derive(Serialize, Clone)]
pub struct ImageMetadata {
    pub image_data: String,
    pub exif_data: String,
    pub width: u32,
    pub height: u32,
    pub aspect_ratio: String,
    pub format: String,
    pub color_depth: Option<u8>,
}

#[derive(Serialize)]
pub struct AiDetectionResult {
    pub is_ai_generated: bool,
    pub format: String,
}
